### PR TITLE
remove hard code http scheme of short url #4656

### PR DESCRIPTION
**Pull request by ripoul**
remove the hard code http. get the scheme of the request and use it to create the new url

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -748,9 +748,10 @@ def shortner(self):
         obj = models.Url(url=url)
         db.session.add(obj)
         db.session.commit()
+        scheme = request.scheme
         return Response(
-            'http://{request.headers[Host]}/{directory}?r={obj.id}'.format(
-                request=request, directory=directory, obj=obj),
+            '{scheme}://{request.headers[Host]}/{directory}?r={obj.id}'.format(  
+            scheme = scheme, request=request, directory=directory, obj=obj),
```
Comment (xrmx) [inline superset/views/core.py]: no spaces around =, and don't  add whitespace before new line please

**Review by xrmx — COMMENTED** [review]
Wondering if we can use flask's url_for instead of building the url manually

**Comment by ripoul** [comment]
something like this ? (I never use url_for)

```
url_for(directory,  r=obj.id)
```

**Comment by xrmx** [comment]
@ripoul i think it would be more complicated than that to have the full url with scheme and host. Don't mind that, we can cleanup that later.

**Comment by ripoul** [comment]
how can I make the test succeed ?

**Comment by xrmx** [comment]
@ripoul you fix what hey report:
./superset/views/core.py:754:13: E122 continuation line missing indentation or outdented

that means that you have to readd the 4 spaces indentation you removed )

**Comment by ripoul** [comment]
@xrmx done ! ty for your help ! :smiley:

File: superset/views/core.py [hunk 2]
```diff
@@ -748,9 +748,10 @@ def shortner(self):
         obj = models.Url(url=url)
         db.session.add(obj)
         db.session.commit()
+        scheme = request.scheme
```
Comment (john-bodley) [inline superset/views/core.py]: Why do we need this temporary variable? I sense it adds for more obfuscation and thus would include this on line 754.

**Review by john-bodley — COMMENTED** [review]
Would you mind elaborating in the PR description as to why/when the hard coded “http” string is problematic? This would help provide reviewers with additional context.

**Comment by ripoul** [comment]
for your first question : it's a probleme when your server accept only https request. In this case we would like to respond with the same scheme than the request to be sure the server can handle it.

**Pull request merged**